hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/processors/StreamToStreamJoinPInnerTest.java

```diff
@@ -65,6 +72,67 @@
 @RunWith(HazelcastSerialClassRunner.class)
 public class StreamToStreamJoinPInnerTest extends JetTestSupport {
 
+    /**
+     * An output checker that will consider the actual and expected object lists
+     * equal if they both contain the same items, in any order. If some item is
+     * expected multiple times, it must also be present the same number of times
+     * in the actual output.
+     * <p>
+     * When restoring from the snapshot, a different mechanism is used to
+     * compare watermarks. The WMs must be monotonic, but not strictly monotonic
+     * - after a restore the same WM can be emitted that was already emitted
+     * before restore. The highest actual WM can be lower than the highest
+     * expected one, but not higher - again, after a restore, it might go back.
+     */
+    public static final BiPredicate<List<?>, List<?>> SAME_ITEMS_ANY_ORDER_EQUIVALENT_WMS =
```

Comment:
I don't like the global variable. You could have a static method with `testMode` parameter which would return the function. Dependency would be much cleaner.

Reply:
`testMode` can be protected field in `TestSupport`

Reply:
It's a specific predicate for a specific test, I don't see a reason to generalize it

Reply:
the reason is not use global variable (thread locals are even worse that ordinary globals)

Reply:
by global variable I meant `TEST_CONTEXT`